projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

```diff
@@ -1602,10 +1610,48 @@
 					}, 150 );
 				}
 			} );
+			var currentSlideObj =
+				options.startIndex > 0 ? carousel.slides[ options.startIndex ] : carousel.slides[ 0 ];
+			var currentSlideImage =
+				currentSlideObj && currentSlideObj.el ? currentSlideObj.el.querySelector( 'img' ) : null;
+			var isCurrentSlideImageLoaded =
+				currentSlideImage && currentSlideImage.complete && currentSlideImage.naturalHeight !== 0;
+
+			domUtil.emitEvent( carousel.overlay, 'jp_carousel.afterOpen' );
```

Comment:
I'm wondering if the issue is that this line is no longer in a callback passed to `domUtil.fadeIn`, so the escape key can be pressed before the fade in has completed?

Reply:
Thanks for going deeper here. That's a good place to start looking.

> I'm not sure, but I'm wondering if there's a potential race condition with the event handlers for opening and closing?

I'm nearly losing track of all the events we're firing off. :) I'm not I'm keen to add to the complexity, so might treat this PR as a learning exercise if we can't get it working.